### PR TITLE
Add unit tests and documentation

### PR DESCRIPTION
**Pull request by gavD**
* Add documentation README.md which explains angular-scaffold's usage
* Add sample project illustrating basic usage of angular-scaffold
  * Includes a very simple Express API so users can get up and running immediately
* Add package.json
  * Prepares the project for NPM
  * Supplies dependencies for the unit tests
* Get the unit tests running with Gulp
* Upgrade unit tests to Jasmine 2.0

Test script
--

* Go to https://github.com/radify/angular-scaffold/tree/docs
* Read the documentation
  * **Assert that** it makes sense
* Clone the project and `git checkout docs`
  * Install and run the sample API as per [the documentation](https://github.com/radify/angular-scaffold/tree/docs)
  * **Assert that** you can run the sample API and it works as you would expect

Time spent
--

8 hours

### CODE REVIEW

**Comment by gavD** [comment]
It's probably worth comparing #3 before merging this... The versions of `angular-scaffold` I'm using are from a specific project

**Comment by nateabele** [comment]
Looks great. I have nothing to add other than maybe change the sample app nomenclature to something generic like 'posts'.

**Comment by gavD** [comment]
on it

**Comment by gavD** [comment]
OK, done. To get this completed, we need to:

1. [ ] Document angular-model (I'll start on documentation for angular-model shortly)
1. [ ] Add angular-model into npm/bower
1. [ ] Remove sample-project/tmp/angular-model.js in favour of package managed version (as above)
1. [ ] Merge this PR
1. [ ] Add angular-scaffold into npm/bower
1. [ ] Remove sample-project/tmp/* in favour of package managed version (as above)

We should also consider changing the ur namespace to radify. What do you think?

**Comment by nateabele** [comment]
Agreed on all counts. Let me know if you need any other code changes reviewed.

**Comment by gavD** [comment]
Angular Model is now documented in a PR https://github.com/radify/angular-model/pull/43 - once that's merged we can move forwards with this as well

**Comment by gavD** [comment]
Please see also #8 which adds API docs and switches to Gulp

**Comment by gavD** [comment]
@nateabele I've merged all the changes into this integration branch. If you are happy with this, please merge and pass over to @jfmercer for release to NPM.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    {
+      "name": "Gavin Davies",
+      "email": "gavin@radify.io",
+      "web": "http://radify.io"
+    },
+    {
+      "name": "Warren Seymour",
+      "email": "warren@radify.io",
+      "web": "http://radify.io"
```
Comment (nateabele) [inline package.json]: Warren is listed twice here.

**Comment by nateabele** [comment]
Found one copypasta issue on `package.json`. Everything else looks ace. Feel free to :ship: when that's fixed.

File: package.json (hunk 1 above)
Comment (gavD) [inline package.json]: that's 'cos he's twice as good

**Comment by gavD** [comment]
fixed, merging now. @jfmercer this needs to be released once angular-model is released.

**Comment by jfmercer** [comment]
@gavD `angular-model` has already been released/published to NPM. Can I publish `scaffold` too?

**Comment by gavD** [comment]
Yes please

On Mon, Nov 30, 2015 at 4:53 PM, John Mercer <notifications@github.com>
wrote:

> @gavD <https://github.com/gavD> angular-model has already been
> released/published to NPM. Can I publish scaffold too?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/radify/angular-scaffold/pull/7#issuecomment-160687789>
> .
>



-- 
Gavin Davies | Automation Engineer; Radify, Inc. | radify.io

**Pull request merged**